### PR TITLE
add categorymanager to filter out entry

### PR DESCRIPTION
**Pull request by ongspxm**

### CODE REVIEW

**Comment by ongspxm** [comment]
resolved conflicts. codacy seems irrelevant. can merge

File: src/main/java/seedu/address/model/category/ContainsCategoryPredicate.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+public class ContainsCategoryPredicate implements Predicate<MajorResumeEntry> {
+    private final Category category;
+
+    public ContainsCategoryPredicate(String keyword) {
+        category = new Category(keyword);
+    }
+
+    @Override
+    public boolean test(MajorResumeEntry entry) {
+        return category.equals(entry.getCategory());
```
[the remark this answers is not in the document]
Comment (ongspxm) [reply src/main/java/seedu/address/model/category/ContainsCategoryPredicate.java]: i am hoping to do all the category checking and tag checking logic in the `CategoryManager` and `TagManager`. So the entry dun really need to know how categories are being compared.

**Comment by anubh-v** [comment]
Looks good to me - just rename `MajorResumeEntry` to `ResumeEntry`.
There are some minor improvements suggested for the docs.

**Comment by ongspxm** [comment]
Yay finally ready to merge

File: docs/DeveloperGuide.adoc [hunk 2]
```diff
@@ -328,6 +333,7 @@ Work in Progress
 
 Work in Progress
 
+>>>>>>> master
```
Comment (jhengy) [inline docs/DeveloperGuide.adoc]: Is this line supposed to be removed? Looks like a merge conflict syntax

**Pull request closed without merging**